src/angularComponent/ngcOmniboxSuggestionsDirective.js

```diff
@@ -1,14 +1,53 @@
+import angular from 'angular';
 import NgcOmniboxSuggestionsController from './ngcOmniboxSuggestionsController.js';
 
-export default function ngcOmniboxSuggestionsDirective() {
+ngcOmniboxSuggestionsDirective.$inject = ['$templateCache'];
+export default function ngcOmniboxSuggestionsDirective($templateCache) {
   return {
     restrict: 'E',
     require: '^^ngcOmnibox',
     scope: true,
     controller: NgcOmniboxSuggestionsController,
     controllerAs: 'suggestions',
     compile(tElement) {
-      tElement.attr('role', 'listbox');
+      const el = tElement[0];
+      el.setAttribute('role', 'listbox');
+
+      const categoryEl = el.querySelector(
+        'ngc-omnibox-suggestion-category, [ngc-omnibox-suggestion-category]'
+      );
+      const itemEl = el.querySelector(
+        'ngc-omnibox-suggestion-item, [ngc-omnibox-suggestion-item]'
+      );
+
+      if (categoryEl) {
+        const categoryContainer = angular.element('<div></div>')[0];
+        categoryContainer.setAttribute('ng-repeat',
+            'suggestion in suggestion.children || omnibox.suggestions');
+        categoryEl.setAttribute('ng-if', 'suggestion.children');
+
+        const itemChildrenEl = angular.element('<div></div>')[0];
+        itemChildrenEl.setAttribute('ngc-omnibox-suggestion-item', '');
+        itemChildrenEl.setAttribute('ng-repeat', 'suggestion in suggestion.children');
+        itemChildrenEl.setAttribute('ng-include', '\'category-tmpl\'');
+        if (itemEl.hasAttributes()) {
+          for (let i = 0; i < itemEl.attributes.length; i++) {
+            const attr = itemEl.attributes[i];
+            itemChildrenEl.setAttribute(attr.name, attr.value);
+          }
+        }
+        itemEl.parentNode.appendChild(itemChildrenEl);
+
+        categoryContainer.appendChild(categoryEl);
+        categoryContainer.appendChild(itemEl);
```

Comment:
You've built a DOM structure in this code, I think it would help if there were a comment that contained the HTML representation of the DOM structure that is being built by this JavaScript. Of course, we would need to keep the HTML in comment 100% strictly up to date with the JavaScript because obsolete comments are worse than no comments at all...